src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { FullTodo } from './types/FullTodo';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todosWithUsers: FullTodo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<FullTodo[]>(todosWithUsers);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedUserId, setNewUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const getMaxId = (todosArray: FullTodo[]): number => {
+    const copy = [...todosArray].sort((prev, curr) => curr.id - prev.id);
+
+    return copy[0].id + 1;
+  };
+
+  const addNewTodo = (title: string, userId: number) => {
+    const newTodo: FullTodo = {
+      id: getMaxId(todos),
+      title,
+      completed: false,
+      userId,
+      user: getUserById(selectedUserId),
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (newTitle && selectedUserId) {
+      addNewTodo(newTitle, selectedUserId);
+      setNewTitle('');
+      setNewUserId(0);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Consider using a custom hook to handle form validation instead of using `useEffect` to improve readability and maintainability. 🎣